packages/flutter/test/material/theme_data_test.dart

```diff
@@ -268,6 +268,40 @@ void main() {
     expect(theme.applyElevationOverlayColor, isTrue);
   });
 
+  testWidgets('splashFactory is InkSparkle only for Android non-web when useMaterial3 is true', (WidgetTester tester) async {
+    final ThemeData theme = ThemeData.fallback().copyWith(useMaterial3: true);
+
+    // Basic check that this theme is in fact using material 3.
+    expect(theme.useMaterial3, true);
+
+    switch (debugDefaultTargetPlatformOverride!) {
+      case TargetPlatform.android:
+        expect(theme.splashFactory, equals(InkSparkle.splashFactory));
+        break;
+      case TargetPlatform.iOS:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+      case TargetPlatform.macOS:
+      case TargetPlatform.windows:
+        expect(theme.splashFactory, equals(InkSplash.splashFactory));
+    }
+  }, skip: kIsWeb, variant: TargetPlatformVariant.all());
+
+  testWidgets('splashFactory is InkSplash for every platform scenario, including Android non-web, when useMaterial3 is false', (WidgetTester tester) async {
+    // Currently, useMaterial3 defaults to false.
+    final ThemeData theme = ThemeData.fallback();
```

Comment:
Even though the default `useMaterial3` is false, we will be changing this in the future. It would be better to be explicit about this here:

```suggestion
    final ThemeData theme = ThemeData(useMaterial3: false);
```